### ISSUE TITLE
Plugin fails to create .output/photos folder on Windows

### ISSUE THREAD

**Issue by antonmosich** [pelican-plugins/photos#29]
Hey, I was just trying to use your plugin, and when trying to generate the articles I was testing with, photos does not create a photos folder and subsequently does not process the images either.
I am using Windows 10.
I tried to, but could not replicate the error on a Live Linux system, which leads me to believe, that the problem is due to my OS.

When I try to generate the example from the repo, I get the following log:
```(pelican) D:\Development\selfhost\pelican-plugins\photos\examples>pelican
[13:05:58] WARNING  Watched path does not exist:                                                              log.py:91
                    D:\Development\selfhost\pelican-plugins\photos\examples\content\images
<class 'pelican.plugins.photos.photos.CaptionCache'>
<class 'pelican.plugins.photos.photos.ExifCache'>
<class 'pelican.plugins.photos.photos.ExcludeCache'>
<class 'pelican.plugins.photos.photos.CaptionCache'>
<class 'pelican.plugins.photos.photos.ExifCache'>
<class 'pelican.plugins.photos.photos.ExcludeCache'>
<class 'pelican.plugins.photos.photos.CaptionCache'>
<class 'pelican.plugins.photos.photos.ExifCache'>
<class 'pelican.plugins.photos.photos.ExcludeCache'>
<class 'pelican.plugins.photos.photos.CaptionCache'>
<class 'pelican.plugins.photos.photos.ExifCache'>
<class 'pelican.plugins.photos.photos.ExcludeCache'>
<class 'pelican.plugins.photos.photos.CaptionCache'>
<class 'pelican.plugins.photos.photos.ExifCache'>
<class 'pelican.plugins.photos.photos.ExcludeCache'>
<class 'pelican.plugins.photos.photos.CaptionCache'>
<class 'pelican.plugins.photos.photos.ExifCache'>
<class 'pelican.plugins.photos.photos.ExcludeCache'>
<class 'pelican.plugins.photos.photos.CaptionCache'>
<class 'pelican.plugins.photos.photos.ExifCache'>
<class 'pelican.plugins.photos.photos.ExcludeCache'>
[13:05:59] WARNING  photos: 'PHOTO_SQUARE_THUMB'                                                              log.py:91
Done: Processed 5 articles, 0 drafts, 0 hidden articles, 0 pages, 0 hidden pages and 0 draft pages in 1.51 seconds.```

**Comment by phibos** [comment]
The directory should be created at this point in the code https://github.com/pelican-plugins/photos/blob/2a10a1a7a1c1c363c52c9affe865369621e20a52/pelican/plugins/photos/photos.py#L693

Does it raise an exception?

Can you please set
```python
PHOTO_RESIZE_JOBS = -1
```
this will start the process with more debugging. And please try to start pelican with ```-D``` to enable debugging.

**Comment by Flugschwein** [comment]
Okay, I tried around a bit, and the error seems to be connected to the multiprocessing module. The generation worked once I set the `PHOTO_RESIZE_JOBS = -1`.
Though, using the `-D` flag and `PHOTO_RESIZE_JOBS = 1`, the program _did_ raise an exception:
```
           ERROR    photos: 'PHOTO_WATERMARK'                                                                                                                    log.py:96
           WARNING  photos: An exception occurred                                                                                                                log.py:91
                    multiprocessing.pool.RemoteTraceback:
                    """
                    Traceback (most recent call last):
                      File "C:\Users\Anton\anaconda3\envs\pelican\lib\multiprocessing\pool.py", line 125, in worker
                        result = (True, func(*args, **kwds))
                      File "C:\Users\Anton\anaconda3\envs\pelican\lib\site-packages\pelican\plugins\photos\photos.py", line 687, in process
                        image = func(image, *operation_args, **operation_kwargs)
                      File "C:\Users\Anton\anaconda3\envs\pelican\lib\site-packages\pelican\plugins\photos\photos.py", line 775, in _operation_watermark
                        if not pelican_settings["PHOTO_WATERMARK"]:
                    KeyError: 'PHOTO_WATERMARK'
                    """

                    The above exception was the direct cause of the following exception:

                    KeyError: 'PHOTO_WATERMARK'
```
That leads me to believe, that the multiprocessing somehow messes up that global variable...

**Comment by EddyXorb** [comment]
I can confirm that the exact same pelican blog repository does not work with the photos plug-in under windows, but in Linux it does. I am talking about version 1.3.0.

**Comment by phibos** [comment]
This issue should have been fixed with PR #74 feel free to test. It will be included in the next 1.5.0 release.
